kubernetes-client/src/main/java/io/fabric8/kubernetes/client/utils/Serialization.java

```diff
@@ -152,6 +152,8 @@ public static ObjectMapper yamlMapper() {
     String specFile = readSpecFileFromInputStream(is);
     if (containsMultipleDocuments(specFile)) {
       return (T) getKubernetesResourceList(parameters, specFile);
+    } else if (specFile.contains(DOCUMENT_DELIMITER)) {
```

Comment:
This statement will match things like
```yaml
metadata:
  annotations:
    annotation: ---
```

Reply:
A quick `replaceAll("^---.*$", "")` should probably do the job.

Reply:
We should also be careful with scalars, not sure how this might be interpreted in those cases.

Reply:
I tried with `replaceAll("^---.*$", "")` but it doesn't seem to replace document delimiters. I checked and yaml string seems to be in this format:

![Screenshot from 2022-01-07 15-05-52](https://user-images.githubusercontent.com/13834498/148524252-603339ec-76d0-42f1-b5c0-8e67440978f8.png)

`specFile.replaceAll(System.lineSeparator() + "---", "")` seems to work. Shall I update to use this instead?

Reply:
So basically if it`s a single document YAML it can only have line separators:
- beginning
- ending
- both
- none

So these replacements might be applicable:
- For optional beginning delimiter:
  `^---\s?.*?\r?\n`
- For optional ending delimiter:
  `\n---\s?.*?\r?\n?$`

Reply:
So after our chat on Friday, it all came down to 
```
      specFile = specFile.replaceAll("^---([ \\t].*?)?\\r?\\n","");
      specFile = specFile.replaceAll("\\n---([ \\t].*?)?\\r?\\n?$","\n");
```

However, I'm checking the spec and this is probably incorrect.
https://yaml.org/spec/1.0/#id2561718

Reply:
So, it's probably better to leave it like this for now, and account for further edge cases in the future.